pkg/kubelet/network/plugins.go

```diff
@@ -38,6 +38,13 @@ type NetworkPlugin interface {
 	// before any other methods are called.
 	Init(host Host) error
 
+	// Tells the plugin to start any long-running operations; the plugin
+	// should not block this function
+	Run() error
+
+	// Indicates to the plugin that the pod CIDR changed, if controlled by kubelet
+	PodCIDRChanged(cidr string)
```

Comment:
Under what conditions do we expect this to actually get called, and what do we expect the plugin to do?  Is it realistic to expect the Kubelet to be graceful about changing the PodCIDR on the fly?  I'd be fine with simply logging "Pod CIDR has changed, but Pods exist; changes will be enacted when possible".  We could even punt completely and say that PodCIDR needs a node reboot to take effect.  If it is a real issue for people we can come back to it later.